plugin/driver_test.go

```diff
@@ -320,33 +320,34 @@ func TestFunctional_cases(t *testing.T) {
 			unveilDefaults: false,
 			exp:            &drivers.ExitResult{ExitCode: 2},
 		},
-		// TODO(shoenig) re-enable these once #29 is fixed.
-		//
 		// write to task directory
-		// {
-		// 	name:           "write to task directory",
-		// 	user:           "nomad-80000",
-		// 	command:        "cp",
-		// 	unveilDefaults: true,
-		// 	args:           []string{"/etc/hosts", "${NOMAD_TASK_DIR}"},
-		// 	exp:            &drivers.ExitResult{ExitCode: 0},
-		// },
-		// {
-		// 	name:           "write to alloc directory",
-		// 	user:           "nomad-80000",
-		// 	command:        "cp",
-		// 	unveilDefaults: true,
-		// 	args:           []string{"/etc/hosts", "${NOMAD_ALLOC_DIR}"},
-		// 	exp:            &drivers.ExitResult{ExitCode: 0},
-		// },
-		// {
-		// 	name:           "write to secrets directory",
-		// 	user:           "nomad-80000",
-		// 	command:        "cp",
-		// 	unveilDefaults: true,
-		// 	args:           []string{"/etc/hosts", "${NOMAD_SECRETS_DIR}"},
-		// 	exp:            &drivers.ExitResult{ExitCode: 0},
-		// },
+		{
+			name:           "write to task directory",
+			user:           "nomad-80000",
+			command:        "sh",
+			unveilDefaults: true,
+			unveilPaths:    []string{"r:/etc/hosts"},
+			args:           []string{"-c", "cp /etc/hosts ${NOMAD_TASK_DIR}"},
+			exp:            &drivers.ExitResult{ExitCode: 0},
+		},
+		{
+			name:           "write to alloc directory",
+			user:           "nomad-80000",
+			command:        "sh",
+			unveilDefaults: true,
+			unveilPaths:    []string{"r:/etc/hosts"},
+			args:           []string{"-c", "cp /etc/hosts ${NOMAD_ALLOC_DIR}"},
+			exp:            &drivers.ExitResult{ExitCode: 0},
+		},
+		{
+			name:           "write to secrets directory",
+			user:           "nomad-80000",
+			command:        "sh",
+			unveilDefaults: true,
+			unveilPaths:    []string{"r:/etc/hosts"},
+			args:           []string{"-c", "cp /etc/hosts ${NOMAD_SECRETS_DIR}"},
+			exp:            &drivers.ExitResult{ExitCode: 0},
+		},
```

Comment:
It probably seems "obvious" now, but if we had a test that exercised the reverse case (that `NOMAD_*_DIR` is not unveiled when `unveil_defaults=false`), that test would illustrate that it's by design and not by accident.